src/AutoMapper.php

```diff
@@ -157,6 +143,16 @@ public function bindCustomTransformer(CustomTransformerInterface $customTransfor
         $this->customTransformersRegistry->addCustomTransformer($customTransformer);
     }
 
+    public function register(MapperGeneratorMetadataInterface $configuration): void
+    {
+        $this->mapperGeneratorMetadataRegistry->register($configuration);
+    }
+
+    public function getMetadata(string $source, string $target): ?MapperGeneratorMetadataInterface
+    {
+        return $this->mapperGeneratorMetadataRegistry->getMetadata($source, $target);
+    }
```

Comment:
I've refactored a bit, in order to introduce a `MapperGeneratorMetadataRegistry` class, which will be useful to de-complexify this class as well as `MapperMetadata` class